play-akka-http-ws-standalone/src/main/java/play/libs/ws/akkahttp/StandaloneAkkaHttpWSResponse.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright (C) 2009-2017 Lightbend Inc. <https://www.lightbend.com>
+ */
+package play.libs.ws.akkahttp;
+
+import akka.http.javadsl.model.HttpHeader;
+import akka.http.javadsl.model.HttpResponse;
+import akka.http.javadsl.model.ResponseEntity;
+import akka.http.javadsl.model.headers.SetCookie;
+import akka.stream.Materializer;
+import akka.stream.javadsl.Sink;
+import akka.stream.javadsl.Source;
+import akka.util.ByteString;
+import play.libs.ws.BodyReadable;
+import play.libs.ws.StandaloneWSResponse;
+import play.libs.ws.WSCookie;
+import play.libs.ws.WSCookieBuilder;
+
+import java.time.Duration;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+public final class StandaloneAkkaHttpWSResponse implements StandaloneWSResponse {
+
+  // FIXME make configurable
+  final Duration UNMARSHAL_TIMEOUT = Duration.ofSeconds(1);
+
+  private final HttpResponse response;
+  private final Materializer mat;
+
+  private HttpResponse strictResponse;
+
+  StandaloneAkkaHttpWSResponse(HttpResponse response, Materializer mat) {
+    this.response = response;
+    this.mat = mat;
+  }
+
+  /**
+   * @return all the headers from the response.
+   */
+  @Override
+  public Map<String, List<String>> getHeaders() {
+    final Map<String, List<String>> headers = new HashMap<>();
```

Comment:
Should this be a map with case-insensitive ordering?

Reply:
Yes.

See https://github.com/playframework/play-ws/issues/234.